monitoring_scripts/http_monitor.py

```diff
@@ -0,0 +1,113 @@
+"""HTTP monitor script main module"""
+
+import argparse
+import logging
+import re
+import requests
+
+from . import nagios_common as nagios
+
+parser = argparse.ArgumentParser(
+    prog='http-monitor',
+    description='Nagios-compatible plugin to check a resource availability over HTTP/HTTPS. By '
+                'default is considered to be in a OK status if the HTTP status code is 200 and in '
+                'critical status is considered to be in a OK status if the HTTP status code is '
+                '200 and in CRITICAL status otherwise, different behaviours may be specified '
+                'using optional arguments.',
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+parser.add_argument(
+    'url',
+    help='address of resource to monitor (http/https), must be properly formatted (i.e. '
+         'http://sld.tld[:port] or http://127.0.0.1[:port])'
+)
+parser.add_argument(
+    '--timeout',
+    default=30,
+    type=int,
+    help='maximum time (in seconds) to wait for a response, after which the resource is '
+         'considered in CRITICAL status '
+)
+parser.add_argument(
+    '--redirect-unknown',
+    action='store_true',
+    help='if enabled a 302 status code will result in UNKNOWN status. If disabled a 302 status '
+         'code will result in '
+         'CRITICAL status'
+)
+parser.add_argument(
+    '--debug',
+    action='store_true',
+    default=False,
+    help='display debug messages, DO NOT enable in actual Nagios use'
+)
+
+
+def valid_http_url(url):
```

Comment:
This seems a bit overkill to reimplement considering https://docs.python.org/3.5/library/urllib.parse.html could allow you do the same

Reply:
Will use `urlparse` for URL validation, do you want the check to be less strict anyhow?

Reply:
Since it's a configuration setting, I'm not too concerned with the input validation as anything invalid will be up to the user to fix. I'd recommend making the check as lax as https://tools.ietf.org/html/rfc2616#section-3.2.1 (and https://tools.ietf.org/html/rfc2396) would allow for, even if the URI doesn't make sense. Better support a bogus URI than refuse a perfectly compliant URI